packages/api/routes/apikeys/user/deleteKeysByUserMiddleware.js

```diff
@@ -0,0 +1,27 @@
+const db_apikey = require("../../../lib/db.apikey");
+
+// Return a function for deleting all API keys for a given user.
+module.exports = function deleteKeysByUser() {
+  return async (req, res, next) => {
+    const apikey = await db_apikey.attach();
+    const user = req.body.user ? req.body.user.trim() : req.body.user;
+
+    if (user) {
+      const dbRes = await apikey.deleteKeysByUser(user);
+      const doc = dbRes.error
+        ? dbRes
+        : {
+            user: user,
+            message: dbRes.deletedCount + " key(s) deleted."
+          };
+
+      res.send(doc);
+    } else {
+      res.send({
+        error: "Invalid Parameter",
+        message: "Username Missing: Username cannot be empty."
```

Comment:
I suggest changing this to "user id".  The value for `user` can be any string, but it's meant to be the user's UUID.
```suggestion
        message: "user id missing: user id cannot be empty."
```